### PR TITLE
Leaderboard: hit the API

### PR DESCRIPTION
**Pull request by ad9311**
### Greetings! 👋🏽 

In this PR I present these changes:

- When the application is launched for the very first time, the app sends a POST request to the leaderboard API with the name of the game and gest an ID that it's later saved in the localstorage.
- After that, the app will use the ID previously saved in the localstorage to send and receive data.
- When the user adds a new score, it is sent to the API as a POST request.
- When the user clicks the refresh button, the app sends a GET request and gets all the scores previously saved.
- If an error occurs is properly handled and lets the user know to try again.
- Make use of async, await, promises, and setTimeout function.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+    await api.getNewGameID({ name: 'MyGame' });
+    api.getNewGameID().catch(dom.renderAPIMessage);
+    dom.setLocalStorage(api);
+  }
+}
 
-  // Lodash, now imported by this script
-  element.innerHTML = _.join(['Hello', 'webpack'], ' ');
-  element.classList.add('hello');
+async function addNewScore() {
```
Comment (arslanbisharat) [inline src/index.js:17]: - [ ] Please add arrow functions and avoid using current keyboard.

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+  const userData = dom.getUserData();
+  lb.addNewScore(userData);
+  api.getUserData(userData);
+  await api.sendNewScore().catch(dom.renderAPIMessage);
+  dom.cleanFields();
+  dom.renderAPIMessage(api.message);
+}
 
-  return element;
+async function refreshScores() {
```
Comment (arslanbisharat) [inline src/index.js:26]: - [ ] Please add arrow functions and avoid using current keyboard.

**Comment by ad9311** [comment]
As mentioned by the previous reviewer I changed the main functions in **/src/index.js**  to arrow functions.

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**